lib/nga.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+
+function handle_error() {
+    echo "Script exited with status ${2} at line ${1}"
+    if [ -z "${savefiles}" ] && [ -d $tmpprefix ]; then rm -f $tmpprefix*; fi
+    if [ -z "${savefiles}" ]  && [ -d $tmpprefix ]; then rm -f $outprefix*; fi
+}
+
+trap 'handle_error ${LINENO} $?' ERR
+
+# cause the script to fail if any commands exist with non-zero status
+set -e
+
+if [[ -z "$1" || -z "$2" || -z "$3" ]]; then
+    echo "Usage:tiff_to_pyramid.bash <tmpdir> <full path to image source> <full path to output target> <save all working files - any value will do>";
+    exit 1;
+fi
+if [[ ! -d $1 || ! -f $2 ]]; then
+    echo "Either the tmpdir or input file do not exist.";
+    exit 1;
+fi
+
+if [ -f $3 ]; then
+    echo "The output file already exists! Deleting it.";
+    if [ -z "${savefiles}" ]; then rm -f $3; fi
+fi
+
+touch $3;
+
+if [ ! -f $3 ]; then
+    echo "Could not create the output file.";
+    exit 1;
+fi
+
+rm -f $3
+
+tmpdir=$1
+# only process the first page of a potentially multi-page image document
```

Comment:
What does this comment apply to?

Reply:
I have no idea. this is basically a copy of the NGA's script, with minor adjustments to paths & such

Reply:
Mike was very clear about wanting to run the NGA scripts unmodified.   Let's ensure that the rest of our code encapsulates them cleanly, but let's not attempt to modify their internals (even opaque comments) and redirect any questions about how the code functions back to whoever made the scripts at the NGA